### PR TITLE
IGNITE-23776 pendingChangeTriggerKey is updated even if rebalance is skipped

### PR DESCRIPTION
**Pull request by alievmirza**
https://issues.apache.org/jira/browse/IGNITE-23776

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java [hunk 1]
```diff
@@ -203,9 +203,15 @@ public static CompletableFuture<Void> updatePendingAssignmentsKeys(
                                         put(partChangeTriggerKey, longToBytesKeepingOrder(revision))
                                 ).yield(PLANNED_KEY_UPDATED.ordinal()),
                                 iif(value(partAssignmentsPendingKey).eq(partAssignmentsBytes),
-                                        ops(remove(partAssignmentsPlannedKey)).yield(PLANNED_KEY_REMOVED_EQUALS_PENDING.ordinal()),
+                                        ops(
+                                                remove(partAssignmentsPlannedKey),
+                                                put(partChangeTriggerKey, longToBytesKeepingOrder(revision))
```
Comment (ibessonov) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java:208]: Can we do the `longToBytesKeepingOrder(revision)` just once?

Comment (ibessonov) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java:208]: Please update the comment too

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneRebalanceUtil.java [hunk 2]
```diff
@@ -200,9 +202,15 @@ public static CompletableFuture<Void> updatePendingAssignmentsKeys(
                                         put(partChangeTriggerKey, longToBytesKeepingOrder(revision))
                                 ).yield(PLANNED_KEY_UPDATED.ordinal()),
                                 iif(value(partAssignmentsPendingKey).eq(partAssignmentsBytes),
-                                        ops(remove(partAssignmentsPlannedKey)).yield(PLANNED_KEY_REMOVED_EQUALS_PENDING.ordinal()),
+                                        ops(
+                                                remove(partAssignmentsPlannedKey),
+                                                put(partChangeTriggerKey, longToBytesKeepingOrder(revision))
```
Comment (ibessonov) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneRebalanceUtil.java:207]: Same question, basically

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtilUpdateAssignmentsTest.java [hunk 3]
```diff
@@ -521,12 +525,18 @@ private void test(
                     KV_UPDATE_CONTEXT);
         }
 
+        keyValueStorage.put(
+                RebalanceUtil.pendingChangeTriggerKey(tablePartitionId).bytes(),
+                longToBytesKeepingOrder(1),
+                KV_UPDATE_CONTEXT);
+
+
```
Comment (ibessonov) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtilUpdateAssignmentsTest.java:533]:
```suggestion
                KV_UPDATE_CONTEXT
        );

```

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryMsInvokeTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    @ParameterizedTest
+    @MethodSource("assignments")
+    public void testPendingChangeTriggerKey(
+            Set<Assignment> currentPending,
+            Set<Assignment> pending
+    ) throws Exception {
+        if (currentPending != null) {
+            metaStorageManager.put(
+                    pendingPartAssignmentsKey(tablePartitionId), Assignments.toBytes(currentPending, assignmentsTimestamp)
+            ).get();
```
Comment (ibessonov) [inline modules/table/src/test/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryMsInvokeTest.java:94]: Usually we use `assertThat(*, willCompleteSuccessfully())`, it includes the timeout. Waiting without timeout is a bad habit.

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneRebalanceUtil.java (hunk 2 above)
Comment (alievmirza) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneRebalanceUtil.java:207]: done

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtilUpdateAssignmentsTest.java (hunk 3 above)
Comment (alievmirza) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtilUpdateAssignmentsTest.java:533]: updated

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryMsInvokeTest.java (hunk 4 above)
Comment (alievmirza) [reply modules/table/src/test/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryMsInvokeTest.java:94]: refactored

**Review by ibessonov — APPROVED** [review]

**Pull request merged**